google-cloud-spanner/src/test/java/com/google/cloud/spanner/ValueTest.java

```diff
@@ -713,28 +718,56 @@ public void serialization() {
     reserializeAndAssert(Value.bytes(newByteArray("abc")));
     reserializeAndAssert(Value.bytes(null));
 
-    reserializeAndAssert(
-        Value.boolArray(new boolean[] {false, true}));
-    reserializeAndAssert(Value.boolArray(Arrays.asList(true, false)));
+    reserializeAndAssert(Value.boolArray(new boolean[] { false, true }));
+    reserializeAndAssert(Value.boolArray(BrokenSerializationList.of(true, false)));
     reserializeAndAssert(Value.boolArray((Iterable<Boolean>) null));
 
-    reserializeAndAssert(
-        Value.int64Array(Arrays.asList(1L, 2L)));
-    reserializeAndAssert(
-        Value.int64Array(new long[] {1L, 2L}));
+    reserializeAndAssert(Value.int64Array(BrokenSerializationList.of(1L, 2L)));
+    reserializeAndAssert(Value.int64Array(new long[] { 1L, 2L }));
     reserializeAndAssert(Value.int64Array((Iterable<Long>) null));
 
-    reserializeAndAssert(
-        Value.float64Array(new double[] {.1, .2}));
-    reserializeAndAssert(Value.float64Array(Arrays.asList(.1, .2, .3)));
+    reserializeAndAssert(Value.float64Array(new double[] { .1, .2 }));
+    reserializeAndAssert(Value.float64Array(BrokenSerializationList.of(.1, .2, .3)));
     reserializeAndAssert(Value.float64Array((Iterable<Double>) null));
 
-    reserializeAndAssert(
-        Value.stringArray(Arrays.asList("a", "b")));
+    BrokenSerializationList<String> of = BrokenSerializationList.of("a", "b");
+    reserializeAndAssert(Value.stringArray(of));
     reserializeAndAssert(Value.stringArray(null));
 
     reserializeAndAssert(
-        Value.bytesArray(Arrays.asList(newByteArray("a"), newByteArray("b"))));
+        Value.bytesArray(BrokenSerializationList.of(newByteArray("a"), newByteArray("b"))));
```

Comment:
I think we had discussed that in certain cases Value does not create a copy so we might need to handle those cases in Value class.